### DISCUSSION TITLE
tls: failed to verify certificate: x509: certificate is valid for *. traefik. default, not docker. io.

### DISCUSSION THREAD

**Discussion by H-TWINKLE** [k3s-io/k3s#10697] [Q&A]
I built a private repository using Nexus and it is located on a machine in the K3S cluster. 
I created a private certificate for Docker.io using OpenSSL and then used nginx's HTTPS proxy to access the Nexus repository.
When I use Docker to pull the image or WGET to access it https://docker.io When using a private repository, error info by :  

    tls: failed to verify certificate: x509: certificate is valid for *. traefik. default, not docker. io. 

How can I solve this problem.

**Comment by brandond** [comment]
> `certificate is valid for *. traefik. default`

It sounds like you're hitting the default Traefik certificate. On K3s nodes, the Traefik ingress controller service occupies ports 80 and 443 by default. How did you intend to expose nexus outside the cluster?

> I created a private certificate for Docker.io using OpenSSL and then used nginx's HTTPS proxy to access the Nexus repository.

That's not the right thing to do at all. What steps are you following here?
